### PR TITLE
chore: decomposing action execution flow methods into smaller methods 

### LINKED ISSUE #18760: [Task]:  Decomposition of action execution flow to support method overloads
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

Context: For datasource environments project, the action execution flow needs some methods are required to be overloaded in the enterprise edition code. However, since the methods are large, for maintenance purposes the difference in EE and CE versions of code should be minimum. Hence the decomposition of code into relatively smaller  methods.

- Break down execute action methods for improved readability

### PR DESCRIPTION
**Pull request by sondermanish**
## Description

> To accommodate for datasource context adaption for environments,  the execute action flow needs to be changed so that the action execution flow in EE introduce minimal code diffs. Currently the executeAction method implicitly  handles `dsContext` fetch. this method requires some rework for EE, To avoid a parallel set of method to address the` dsContext fetch method` rework these changes were required.

> TL; DR : This Pr intends to streamline and improve code readability of action execution flow in `executeAction methods` by breaking the main executeAction methods into smaller methods.

Fixes #18760  

## Type of change
 > Chore

## How Has This Been Tested?
- Manual
- Jest

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/NewActionServiceCEImpl.java [hunk 1]
```diff
@@ -692,13 +690,73 @@ public Mono<ActionExecutionResult> executeAction(ExecuteActionDTO executeActionD
...
                         pluginExecutorMono,
-                        pluginMono
-                )
+                        pluginMono,
+                        actionMono
+                );
+
+    }
+
+    protected Mono<ActionExecutionResult> getExecutionMonoFromValidatedDatasourceMono(ExecuteActionDTO executeActionDTO, Mono<Datasource> validatedDatasourceMono, Plugin plugin, PluginExecutor pluginExecutor, ActionDTO action, Datasource datasource) {
```
Comment (sondermanish) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/NewActionServiceCEImpl.java:703]: This could be private as well, as this wouldn't be called from the EE version

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/NewActionServiceCE.java [hunk 2]
```diff
@@ -38,6 +38,8 @@ public interface NewActionServiceCE extends CrudService<NewAction, String> {
     Mono<ActionExecutionResult> executeAction(ExecuteActionDTO executeActionDTO);
 
     Mono<ActionExecutionResult> executeAction(Flux<Part> partsFlux, String branchName);
+
+    Mono<ActionExecutionResult> executeAction(Flux<Part> partsFlux, String branchName, String environmentName);
```
Comment (sondermanish) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/NewActionServiceCE.java:42]: Created a new one instead of replacing just for test cases.

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/NewActionServiceCEImpl.java [hunk 3]
```diff
@@ -645,132 +640,150 @@ public Mono<ActionExecutionResult> executeAction(ExecuteActionDTO executeActionD
...
         }
+    }
 
-        String actionId = executeActionDTO.getActionId();
-        AtomicReference<String> actionName = new AtomicReference<>();
-        // Initialize the name to be empty value
-        actionName.set("");
-        // 2. Fetch the action from the DB and check if it can be executed
-        Mono<NewAction> actionMono = repository.findById(actionId, actionPermission.getExecutePermission())
+    protected Mono<NewAction> getAction(String actionId) {
```
Comment (nidhi-nair) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/NewActionServiceCEImpl.java:645]: Hey can we please name these functions in a way that will help a new contributor immediately understand what's up? getExecutableAction in this instance perhaps. Also add a one liner as docs so we know what we're using when we call this. I say this because our repo layer has similar APIs that function as straight forward DB calls and this addition might get confusing at the first glance. We should always try to make sure that consumers know exactly what they're using without having to read code.

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/NewActionServiceCEImpl.java [hunk 4]
```diff
@@ -645,132 +640,150 @@ public Mono<ActionExecutionResult> executeAction(ExecuteActionDTO executeActionD
...
+                                                                          PluginExecutor pluginExecutor,
+                                                                          String environmentName) {
+
+        Mono<Datasource> validatedDatasourceMono = authenticationValidator.validateAuthentication(datasource).cache();
+
+        Mono<ActionExecutionResult> executionMono = validatedDatasourceMono
+                .flatMap(datasource1 -> {
+                    if (plugin.isRemotePlugin()) {
+                        return datasourceContextService.getRemoteDatasourceContext(plugin, datasource1);
+                    } else {
```
Comment (nidhi-nair) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/NewActionServiceCEImpl.java:716]: Weren't we supposed to pull this out? Is this WIP?

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/NewActionServiceCEImpl.java (hunk 3 above)
Comment (sondermanish) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/NewActionServiceCEImpl.java:645]: modifying as suggested. further code clean up in progress

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/NewActionServiceCEImpl.java [hunk 5]
```diff
@@ -645,132 +640,150 @@ public Mono<ActionExecutionResult> executeAction(ExecuteActionDTO executeActionD
...
                 .flatMap(actionDTO -> datasourceService.getValidDatasourceFromActionMono(actionDTO, datasourcePermission.getExecutePermission()))
                 .cache();
+    }
 
-        Mono<Plugin> pluginMono = datasourceMono
+    protected Mono<Plugin> getPlugin (Mono<Datasource> datasourceMono, String actionId) {
+        return datasourceMono
                 .flatMap(datasource -> {
-                    // For embedded datasources, validate the datasource for each execution
+                    // For embedded datasource, validate the datasource for each execution
```
[the remark this answers is not in the document]
Comment (sondermanish) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/NewActionServiceCEImpl.java:681]: moved the snipped with the `getDatasource` method itself now renamed to `getCachedDatasourceforactionExecution`.

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/NewActionServiceCE.java (hunk 2 above)
Comment (nidhi-nair) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/NewActionServiceCE.java:45]: @sondermanish when are we changing this?

**Review by nidhi-nair — APPROVED** [review]
